### DISCUSSION TITLE
spdyn: badauth :(

### DISCUSSION THREAD

**Discussion by tumichnix** [qdm12/ddns-updater#193] [Q&A] [answered]
Hi,

my spdyn configuration looks like this:

`{
  "settings": [
    {
      "provider": "spdyn",
      "domain": "spdns.de",
      "host": "XXX",
      "user": "XXX",
      "password": "XXX",
      "token": "XXXX-XXXX-XXXX",
      "ip_version": "ipv4",
      "provider_ip": true
    }
  ]
}`

The update request always returns a 401 (badauth). The access data have been checked several times. 
Any suggestions?

**Comment by qdm12** [comment]
Maybe try removing the token from your settings?

What error do you get in the docker logs?

**Reply by tumichnix** [reply]
`2021/04/08 17:33:56 INFO IPv4 address of XXX.spdns.de is 79.235.XXX.XXX and your IPv4 address is 93.228.XXX.XXX                                                                                  
2021/04/08 17:33:56 INFO Updating record [domain: spdns.de | host: XXX | provider: Spdyn] to use 93.228.XXX.XXX                                                                                
2021/04/08 17:33:56 ERROR bad HTTP status: 401: badauth`

It doesn't matter whether the token is empty or not.

**Reply by qdm12** [reply]
From https://wiki.securepoint.de/SPDyn/Variablen

It seems they changed their password url parameter key from `passport` to `pass` I'll fix it tomorrow.

**Reply by qdm12** [reply]
Ah actually it's my browser that auto translated from german `pass` to english `passport` so the implementation was bugged since the beginning. I just fixed it now, it should be available by re-pulling the latest image. Thanks for your patience!

**Comment by qdm12** [comment, marked as answer]
Implementation bug fixed in https://github.com/qdm12/ddns-updater/commit/34623e1b8191152234d69f3f2f0442f9e55d19a7

Thanks for letting me know.

**Reply by tumichnix** [reply]
Mow it works!
Will keep an eye on it for a few days and give feedback if necessary.
Thanks for the good work and support!
